pkg/ebpf/event.go

```diff
@@ -60,10 +64,103 @@ func (t *ConnTuple) copy() *ConnTuple {
 	}
 }
 
+func connTupleFromConn(conn net.Conn, pid uint32) (*ConnTuple, error) {
+	saddr := conn.LocalAddr()
+	shost, sportStr, err := net.SplitHostPort(saddr.String())
+	if err != nil {
+		return nil, err
+	}
+	sport, err := strconv.Atoi(sportStr)
+	if err != nil {
+		return nil, err
```

Comment:
It looks like this is only called in tests -- any reason this is in a non-test file?

Reply:
I can move it, but the functionality isn't necessarily specific to testing. Conversion functions like these, I prefer to have next to either the source or destination type.

Reply:
If this were non-test code intended to be part of the agent, I would say we should we should improve it.  These kinds of conversions tend to be called in a loop, and converting saddr/daddr to string just to call net.SplitHostPort is wasteful compared to switching on the type of saddr.  

That said, it's hard to imagine a scenario where we need to convert a bunch of net.Conn's into a ConnTuple so I'm of a mixed mind.

Reply:
I improved this code to be less wasteful, can you take another look?